WordPress/gutenberg
# Site Editor: Add ability to focus only on editing page content

## What?
Requires https://github.com/WordPress/gutenberg/pull/50082.

Part of https://github.com/WordPress/gutenberg/issues/49404 and https://github.com/WordPress/gutenberg/issues/44461.

Adds functionality to the Site Editor that lets you, when editing a page, switch between focusing on editing the page's content versus editing the page's template.

## Why?
Adding some separation helps to prevent a very common point of frustration encountered when a user attempts to change something on a specific page but the change is made everywhere.

## How?
- Add breadcrumb and sidebar UI to `wp-edit-site` that lets the user toggle between focusing on editing the template and focusing on editing the page.
- Enhance the existing "content locking" functionality in `wp-block-editor`.
  - Allow `templateLock: 'contentOnly'` to be provided at the root level via editor settings.
  - Allow explicitly stating which blocks are consisted to be "content" using `settings.contentBlockTypes`.
  - Allow modification of blocks that are nested within a content block.
- Various UI affordances.
  - Notifications to nudge the user to edit the template when they click on a locked block.
  - A custom right sidebar for when the focus is on editing the page.
  - Display editable content blocks in List View when a block is content locked.

## Testing Instructions
1. Ensure that you have a primary Navigation menu saved.
1. Go to _Appearance → Editor → Navigation_ and select a page.
1. Test the new switching functionality.
1. Check that navigating to _Appearance → Editor → Templates_ and selecting a template results in the same site editor experience as before (no switching functionality).
1. Check that the existing `templateLock: 'contentOnly'` functionality still works by inserting a block pattern that uses it, e.g. this one: https://gist.github.com/richtabor/ddeea41ced691721318649bea8ce9db8

## Screenshots or screencast 

https://user-images.githubusercontent.com/612155/233555590-9036ad7c-e3ad-47dd-8861-8058bb7168d4.mp4
